rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/IBaseLanguageServerExtensions.java

```diff
@@ -42,16 +49,208 @@
         throw new UnsupportedOperationException();
     }
 
-    @JsonRequest("rascal/locationContents")
-    default CompletableFuture<LocationContent> locationContents(URIParameter lang) {
+    @JsonRequest("rascal/supplyProjectCompilationClasspath")
+    default CompletableFuture<String[]> supplyProjectCompilationClasspath(URIParameter projectFolder) {
         throw new UnsupportedOperationException();
     }
 
-    @JsonRequest("rascal/supplyProjectCompilationClasspath")
-    default CompletableFuture<String[]> supplyProjectCompilationClasspath(URIParameter projectFolder) {
+    // BELOW THE FILESYSTEM SERVICE METHODS:
+
+    @JsonRequest("rascal/filesystem/schemes")
+    default CompletableFuture<String[]> fileSystemSchemes() {
+        throw new UnsupportedOperationException();
+    }
+
+    @JsonRequest("rascal/filesystem/watch")
+    default CompletableFuture<Void> watch(WatchParameters params) throws IOException, URISyntaxException {
+        throw new UnsupportedOperationException();
+    }
+
+    @JsonNotification("rascal/filesystem/onDidChangeFile")
+    default void onDidChangeFile(FileChangeEvent event) { };
+
+    @JsonRequest("rascal/filesystem/stat")
+    default CompletableFuture<FileStat> stat(URIParameter uri) throws IOException, URISyntaxException {
+        throw new UnsupportedOperationException();
+    }
+
+    @JsonRequest("rascal/filesystem/readDirectory")
+    default CompletableFuture<FileWithType[]> readDirectory(URIParameter uri) throws URISyntaxException, IOException {
+        throw new UnsupportedOperationException();
+    }
+
+    @JsonRequest("rascal/filesystem/createDirectory")
+    default CompletableFuture<Void> createDirectory(URIParameter uri) throws IOException, URISyntaxException {
+        throw new UnsupportedOperationException();
+    }
+
+    @JsonRequest("rascal/filesystem/readFile")
+    default CompletableFuture<LocationContent> readFile(URIParameter uri) throws URISyntaxException {
+        throw new UnsupportedOperationException();
+    }
+
+    @JsonRequest("rascal/filesystem/writeFile")
+    default CompletableFuture<Void>  writeFile(WriteFileParameters params) throws URISyntaxException, IOException {
+        throw new UnsupportedOperationException();
+    }
+
+    @JsonRequest("rascal/filesystem/delete")
+    default CompletableFuture<Void>  delete(DeleteParameters params) throws IOException, URISyntaxException {
         throw new UnsupportedOperationException();
     }
 
+    @JsonRequest("rascal/filesystem/rename")
+    default CompletableFuture<Void>  rename(RenameParameters params) throws IOException, URISyntaxException {
+        throw new UnsupportedOperationException();
+    }
+
+    public static class DeleteParameters {
```

Comment:
Are these not build-in to lsp4j? or are there a  mirror of the vscode api? (in that case, add a link to the classes they mirror)

Reply:
nope.